File: 1013-FibonacciNumber/1013-FibonacciNumber.py
# Last updated: 10/4/2025, 5:45:31 PM
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        # Contract: 0 <= n <= 30
        if n < 0 or n > 30:
            raise ValueError("Input must satisfy: 0 <= n <= 30")

        if n == 0:
            return 0
        elif n == 1:
            return 1

        a, b = 0, 1
        for _ in range(2, n + 1):
            a, b = b, a + b

        return b

File: 1013-FibonacciNumber/1013-FibonacciNumber.c
// Last updated: 10/4/2025, 5:45:22 PM
int fib(int n) {
    // LeetCode constraint: 0 <= n <= 30
    if (n == 0) return 0;
    if (n == 1) return 1;

    int a = 0, b = 1, c;
    for (int i = 2; i <= n; i++) {
        c = a + b;
        a = b;
        b = c;
    }
    return b;
}
